plugins/text-freetype2/text-functionality.c

```diff
@@ -141,10 +141,56 @@ void set_up_vertex_buffer(struct ft2_source *srcdata)
 	}
 
 skip_word_wrap:;
+
+	// calculate lines width for alignment
+	len = wcslen(srcdata->text);
+	uint32_t current_line_width = 0;
+	uint32_t current_line_index = 0;
+	if (srcdata->custom_width < 100) {
+		srcdata->max_line_width = 0;
+	} else {
+		srcdata->max_line_width = srcdata->custom_width;
+	}
+	for (uint32_t i = 0; i <= len; i++) {
+		if (srcdata->text[i] == L'\n' || i == len) {
+			if (current_line_width > srcdata->max_line_width) {
+				srcdata->max_line_width = current_line_width;
+			}
+			if (current_line_index < max_aligned_lines) {
+				srcdata->lines_width[current_line_index] =
+				        current_line_width;
+				current_line_index++;
+			}
+			current_line_width = 0;
+		} else {
+			glyph_index = FT_Get_Char_Index(srcdata->font_face,
+				srcdata->text[i]);
+			current_line_width += src_glyph->xadv;
+		}
+	}
+
 	fill_vertex_buffer(srcdata);
 	obs_leave_graphics();
 }
 
+uint32_t get_dx_for_line(struct ft2_source *srcdata, int line_index)
+{
+	if (line_index >= max_aligned_lines) {
```

Comment:
Style:  Remove the curly braces for this `if` statement.  Make sure to leave just one blank line between it and the `switch` statement below.